src/NCronJob/Scheduler/ObservablePriorityQueue.cs

```diff
@@ -0,0 +1,143 @@
+using System.Collections;
+using System.Collections.Specialized;
+using System.Diagnostics.CodeAnalysis;
+
+namespace NCronJob;
+
+internal class ObservablePriorityQueue<TElement> : ObservablePriorityQueue<TElement, (DateTimeOffset NextRunTime, int Priority)> where TElement : JobRun
+{
+    public ObservablePriorityQueue(IComparer<(DateTimeOffset NextRunTime, int Priority)> comparer) : base(comparer)
+    { }
+
+    public void Enqueue([DisallowNull] TElement jobWrapper) =>
+        Enqueue(jobWrapper, (jobWrapper.RunAt ?? DateTimeOffset.UtcNow, (int)jobWrapper.Priority));
+}
+internal class ObservablePriorityQueue<TElement, TPriority> : IEnumerable<TElement>, INotifyCollectionChanged where TPriority : IComparable<TPriority>
+{
+    protected readonly PriorityQueue<TElement, TPriority> PriorityQueue;
+    protected readonly object Lock = new();
```

Comment:
Here we could use the new type as well (`System.Lock`)

Reply:
Using the new approach